datameta/static/js/submit.js

```diff
@@ -524,18 +506,137 @@ DataMeta.submit.renderMetadataHelp = function() {
     });
 }
 
+DataMeta.submit.checkBoxes = function(table, checked) {
+    DataMeta.submit.setLock(true);
+    var checkboxes = document.querySelectorAll("input.datameta-entity-check[data-datameta-type='"+ table +"']")
+    checkboxes.forEach(function(box) {
+        box.checked = checked;
+        DataMeta.submit.unchecked[box.getAttribute('data-datameta-uuid')] = ! checked;
+    })
+
+    DataMeta.submit.submit(true);
+    DataMeta.submit.setLock(false);
+}
+
+DataMeta.submit.deleteSelectedMeta = function() {
+
+    DataMeta.submit.setLock(true);
+
+    // Get checked metadatasets
+    var metadatasetIds = Array.from(document.querySelectorAll("input.datameta-entity-check[data-datameta-type='meta']:checked"), input => input.getAttribute("data-datameta-uuid"));
+
+    // Delete all checked metadatasets
+    fetch(DataMeta.api('rpc/delete-metadatasets'),
+    {
+        method: 'POST',
+        headers: {
+            'Accept': 'application/json',
+            'Content-Type': 'application/json'
+        },
+        body:  JSON.stringify({metadatasetIds})
+    }).then(function(response) {
+        if (response.ok) {
+            return DataMeta.submit.refresh();
+        } else if (response.status==401) {
+            DataMeta.new_alert("<strong>ERROR</strong> Deleting the records failed: Unauthenticated.", "danger")
+        } else if (response.status==403) {
+            DataMeta.new_alert("<strong>ERROR</strong> Deleting the records failed: Access denied.", "danger")
+        } else if (response.status==404) {
+            DataMeta.new_alert("<strong>ERROR</strong> Deleting the records failed: Record not found.", "danger")
+        } else if (response.status==400) {
+            response.json().then(json => {
```

Comment:
Here's another async call which get's executed in parallel to the unlock below.

You may just want to throw an error in all of the > 400 cases that contains the message. For the 400 case I have added an `AnnotatedError` class, see `refresh()` for an example on how I use it:
https://github.com/ghga-de/datameta/blob/cd2523b68289d6a093ad0e0d9f78c009bf11c7b6/datameta/static/js/submit.js#L461-L488